### DISCUSSION TITLE
@vite/client not present in the asset pipeline

### DISCUSSION THREAD

**Discussion by ericchernuka** [ElMassimo/vite_ruby#513] [Q&A] [answered]
I am wondering how I can resolve this error `The asset "localhost.test:3000/vite-dev/@vite/client" is not present in the asset pipeline.` after I add the tags:
```ruby
<%= vite_client_tag %>
<%= vite_react_refresh_tag %>
```
If I navigate to that URL, I get 
<img width="612" alt="image" src="https://github.com/user-attachments/assets/383e91c3-0990-4cb9-98f5-6dd9350828b7">

**Comment by ElMassimo** [comment, marked as answer]
Hi Eric!

This seems like a configuration issue in the application, Rails should not attempt to find the Vite client in the asset pipeline.

To confirm this is the issue, try passing `skip_pipeline: true` to [`vite_client_tag`](https://github.com/ElMassimo/vite_ruby/blob/main/vite_rails/lib/vite_rails/tag_helpers.rb#L9).

If that doesn't fix the problem, then it might be that your asset host is configured incorrectly and Rails doesn't detect it as a URL (for example, check if it's missing the `http` protocol).

**Reply by ericchernuka** [reply]
@ElMassimo You're correct. I had to dig, but we had some logic for ngrok that was causing the asset_host to be missing the `http` protocol (nice guess!). All good. Thanks for the quick response!
